Quadriga/src/main/java/edu/asu/spring/quadriga/service/dictionary/mapper/impl/DictionaryDeepMapper.java

```diff
@@ -28,222 +26,172 @@
 import edu.asu.spring.quadriga.service.dictionary.mapper.IDictionaryDeepMapper;
 import edu.asu.spring.quadriga.service.user.mapper.IUserDeepMapper;
 import edu.asu.spring.quadriga.service.workbench.mapper.IProjectDictionaryShallowMapper;
-import edu.asu.spring.quadriga.service.workbench.mapper.IProjectShallowMapper;
 import edu.asu.spring.quadriga.service.workspace.mapper.IWorkspaceDictionaryShallowMapper;
 
 @Service
 public class DictionaryDeepMapper implements IDictionaryDeepMapper {
 
+    @Autowired
+    private IQuadrigaRoleManager roleManager;
+
+    @Autowired
+    private IDictionaryCollaboratorFactory dictionaryCollaboratorFactory;
+
+    @Autowired
+    private IDictionaryFactory dictionaryFactory;
+
+    @Autowired
+    private ICollaboratorFactory collaboratorFactory;
+
+    @Autowired
+    private IProjectDictionaryShallowMapper projectDictionaryShallowMapper;
+
+    @Autowired
+    private IWorkspaceDictionaryShallowMapper workspaceDictionaryShallowMapper;
+
+    @Autowired
+    private IDictionaryItemsFactory dictionaryItemsFactory;
+
+    @Autowired
+    private IDictionaryItemFactory dictionaryItemFactory;
+
+    @Autowired
+    private IUserDeepMapper userDeepMapper;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IDictionary getDictionaryDetails(DictionaryDTO dictionaryDTO) throws QuadrigaStorageException {
+        IDictionary dictionary = null;

```

Comment:
I would right in the beginning before doing anything else test if dictionaryDTO is null and if so return null. Then you don't need this big if statement later.